radius.go

```diff
@@ -22,22 +27,44 @@ func start_radius_server(db *sql.DB) {
 		// grabbing information
 		// note that we assume username = imsi
 		// optional ip is undefined
-		username := rfc2865.UserName_GetString(r.Packet)
-		// password := rfc2865.UserPassword_GetString(r.Packet)
-		var code radius.Code
-		_, err := db.Query("SELECT *  FROM customers WHERE customers.imsi = ?", username)
+		imsi := rfc2865.UserName_GetString(r.Packet)
+		// query through database to find if this imsi matches some IP
+		// if it does, write back the IP
+		// Reject otherwise
+		rows, err := db.Query("SELECT ip FROM static_ips WHERE imsi = ?", imsi)
+		defer rows.Close()
+		var res *radius.Packet
 		if err != nil {
+			// we should reject the user 
 			log.WithError(err).Error("Cannot find this customer in our db")
-			code = radius.CodeAccessReject
+			res = radius.New(radius.CodeAccessReject, []byte(`res`)) // what should we set the secret?
 		} else {
-			code = radius.CodeAccessAccept
-		}
+			// we should accept the user
+			res = radius.New(radius.CodeAccessAccept, []byte(`res`)) // what should we set the secret?
+			// grab the IP
+			var ip string
+			for rows.Next() {
+				if err1 := rows.Scan(&ip); err1 != nil {
+        			log.Fatal(err1)
+    			}
+			}
 
-	    //code = radius.CodeAccessAccept
-	    //code = radius.CodeAccessReject
+			log.Printf("The IP we found for imsi: %s is IP: %s", imsi, ip)
+			// transform ip from string to IPv4 form
+			IP_found := net.ParseIP(ip)
+			// add the ip to FramedIPAddress attribute of our packet
+			err2 := rfc2865.FramedIPAddress_Add(res, IP_found)
+			if (err2 != nil) {
+				log.Fatal("Cannot add the ip to the packet")
+				// we should handle this situation
```

Comment:
Could this be a todo, formatted `// TODO (name or issue number) ` so it's easy to search for with automation?